rust/hyperlane-core/src/types/checkpoint.rs

```diff
@@ -9,10 +9,10 @@ use crate::{utils::domain_hash, Signable, Signature, SignedType, H160, H256};
 /// An Hyperlane checkpoint
 #[derive(Copy, Clone, Eq, PartialEq, Serialize, Deserialize, Debug)]
 pub struct Checkpoint {
-    /// The mailbox address
+    /// The merkle tree hook address
     pub merkle_tree_hook_address: H256,
-    /// The mailbox chain
-    pub mailbox_domain: u32,
+    /// The merkle tree hook chain
+    pub merkle_tree_hook_domain: u32,
```

Comment:
Sorry I know I was confusing here - but because we've already spun up the testnet4 validators and this changes the serialization format of checkpoints in S3, changing this is a backward incompatible change. It'd require all the validators to have their S3 buckets cleared out, which isn't easy to do atm, so I think we should just stay with `mailbox_domain` to preserve the compatibility with the existing checkpoints signed by validators